### DISCUSSION TITLE
using extra layer to swap ctrl & gui keys

### DISCUSSION THREAD

**Discussion by idavydov** [manna-harbour/miryoku#317] [Show and tell]
As far as I understand, there's no equivalent of `MAGIC_SWAP_CTL_GUI` in zmk.

Since I was not using the extra layer, it was quite easy to repurpose it as a way to quickly swap control and gui keys.  So far it seems to work fine.

My `custom_config.h` [looks like](https://github.com/idavydov/miryoku_zmk/blob/master/miryoku/custom_config.h) this:

<details>
<summary>custom_config.h</summary>

```c
#define MIRYOKU_ALPHAS_QWERTY
#define MIRYOKU_NAV_VI

#define MIRYOKU_LAYER_EXTRA \
&kp Q,             &kp W,             &kp E,             &kp R,             &kp T,             &kp Y,             &kp U,             &kp I,             &kp O,             &kp P,             \
U_MT(LCTRL, A),    U_MT(LALT, S),     U_MT(LGUI, D),     U_MT(LSHFT, F),    &kp G,             &kp H,             U_MT(LSHFT, J),    U_MT(LGUI, K),     U_MT(LALT, L),     U_MT(LCTRL, SQT),   \
U_LT(U_BUTTON, Z), U_MT(RALT, X),     &kp C,             &kp V,             &kp B,             &kp N,             &kp M,             &kp COMMA,         U_MT(RALT, DOT),   U_LT(U_BUTTON, SLASH),\
U_NP,              U_NP,              U_LT(U_MEDIA, ESC),U_LT(U_NAV, SPACE),U_LT(U_MOUSE, TAB),U_LT(U_SYM, RET),  U_LT(U_NUM, BSPC), U_LT(U_FUN, DEL),  U_NP,              U_NP
```

</details>

That's quite a simple customization, but I hope it saves someone a half an hour.

**Comment by manna-harbour** [comment]
Thanks for sharing your config!

For Ctrl / Gui swap, see https://github.com/manna-harbour/miryoku/issues/33.

If you do it with a custom Base / Extra layer as done here, you'd also need to swap the basic mods on all layers, unless you restrict yourself to only activating mods while on Base / Extra. For Extra, you could add duplicates of all of the other layers and modify Extra to use those.
